### PR TITLE
Added additional verification to deployment test

### PR DESCRIPTION
**Pull request by clacroix12**
Jira ticket: https://projects.engineering.redhat.com/browse/OCSQE-66

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

File: tests/ecosystem/deployment/test_deployment.py [hunk 1]
```diff
@@ -15,9 +16,139 @@ def test_deployment():
...
+
+    # Verify cluster is running
+    assert is_cluster_running(config.ENV_DATA['cluster_path'])
+
+    # Verify OCS Operator and Local Storage Operator in Succeeded phase
+    csv = ocp.OCP(kind='csv', namespace=namespace)
+    csvs = csv.get()
+    for item in csvs['items']:
+        name = item['metadata']['name']
+        log.info("Checking status of %s", name)
```
Comment (vavuthu) [inline tests/ecosystem/deployment/test_deployment.py]: use f-strings, same in other places

File: tests/ecosystem/deployment/test_deployment.py [hunk 2]
```diff
@@ -15,9 +16,139 @@ def test_deployment():
...
+    # ocs-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='name=ocs-operator',
+        timeout=timeout
+    )
+    # storagecluster-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-operator',
```
Comment (vasukulkarni) [inline tests/ecosystem/deployment/test_deployment.py]: I think the pod names are still 'rook-ceph' and independent of `storage_cluster_name`

File: tests/ecosystem/deployment/test_deployment.py (hunk 1 above)
Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: We actually specify to let the logger do the formatting for you in our [coding guidelines](https://github.com/red-hat-storage/ocs-ci/blob/master/docs/coding_guidelines.md)

File: tests/ecosystem/deployment/test_deployment.py [hunk 3]
```diff
@@ -15,9 +16,139 @@ def test_deployment():
...
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-rbdplugin-provisioner',
+        resource_count=2,
+        timeout=timeout
+    )
+    # osds
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-osd',
+        resource_count=3,
```
Comment (shylesh) [inline tests/ecosystem/deployment/test_deployment.py]: is this osd pod count ? if Yes, is it fixed to 3 all time ?

File: tests/ecosystem/deployment/test_deployment.py [hunk 4]
```diff
@@ -15,9 +16,139 @@ def test_deployment():
...
+    )
+    required_storage_classes = {
+        f'{storage_cluster_name}-cephfs',
+        f'{storage_cluster_name}-ceph-rbd'
+    }
+    storage_classes = storage_class.get()
+    storage_class_names = {
+        item['metadata']['name'] for item in storage_classes['items']
+    }
+    assert required_storage_classes.issubset(storage_class_names)
```
Comment (shylesh) [inline tests/ecosystem/deployment/test_deployment.py]: Do feel we can add ceph health check as well here ? OR is it already covered elsewhere ?

File: tests/ecosystem/deployment/test_deployment.py (hunk 2 above)
Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: Is that for this specific pod or all of the pods? @vasukulkarni

File: tests/ecosystem/deployment/test_deployment.py (hunk 3 above)
Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: This is OSD pod count yes. I based all resource_counts on the [document](https://docs.google.com/document/d/1iQfIGR1N_Uezzo0gs-pSDFQ1q_oANWZGGIuslhmdQqg/edit) linked in the jira ticket

File: tests/ecosystem/deployment/test_deployment.py (hunk 4 above)
Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: All of the verification steps were taken from the [doc](https://docs.google.com/document/d/1iQfIGR1N_Uezzo0gs-pSDFQ1q_oANWZGGIuslhmdQqg/edit). If we want to add additional checks here we can, but this PR is specifically for matching what is outlined in the doc.

File: tests/ecosystem/deployment/test_deployment.py (hunk 2 above)
Comment (vasukulkarni) [reply tests/ecosystem/deployment/test_deployment.py]: @clacroix12 In this PR's jenkins job, I see all the pods are still prefixed with `rook-ceph`, I did change the cluster_name https://github.com/red-hat-storage/ocs-ci/pull/920

Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: Fixed. We're now reading in the pod selector from `constants`.

File: tests/ecosystem/deployment/test_deployment.py (hunk 4 above)
Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: I thought about it again and the doc actually says to check ceph health from the dashboard (which we aren't doing here) so in the next update I'll add a ceph health check.

Comment (clacroix12) [reply tests/ecosystem/deployment/test_deployment.py]: Done

**Comment by clacroix12** [comment]
> Overall LGTM, I'm just not sure about the various constants for `resource_count`, similarly as it mentioned above.

Yeah, I'm not 100% sure either. I think determining this will take some additional testing. I verified the counts with the [official doc](https://access.redhat.com/documentation/en-us/red_hat_openshift_container_storage/4.2/html-single/deploying_openshift_container_storage/index?lb_target=preview#procedure_verifying-ocs-installation_en-usrhocs) so I think it should be okay for the normal use case. If we are manipulating the pod count somehow we can change this to read in the number of expected pods.

**Comment by vasukulkarni** [comment]
LGTM, I guess right now everyone is running with 3 counts so we are safe in jenkins, and I think you can take it up as follow on PR, we have the ENV defined for osd I guess.

**Pull request merged**